### PR TITLE
get-latest-build and get-build

### PR DESCRIPTION
**Pull request by mmitche**
Quick and dirty implementation of a get-latest-build and get-build operation, given the current constraints of the REST APi

### CODE REVIEW

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/GetBuildOperation.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    {
+        GetBuildCommandLineOptions _options;
+        public GetBuildOperation(GetBuildCommandLineOptions options)
+            : base(options)
+        {
+            _options = options;
+        }
+
+        /// <summary>
+        /// Adds a new channel with the specified name.
```
Comment (riarenas) [inline src/Microsoft.DotNet.Darc/src/Darc/Operations/GetBuildOperation.cs]: Gets the build with the specified ID.

File: src/Microsoft.DotNet.Darc/src/Darc/Options/GetBuildCommandLineOptions.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using CommandLine;
+using Microsoft.DotNet.Darc.Operations;
+
+namespace Microsoft.DotNet.Darc.Options
+{
+    [Verb("get-build", HelpText = "Retrieves the latest build of a repository")]
```
Comment (riarenas) [inline src/Microsoft.DotNet.Darc/src/Darc/Options/GetBuildCommandLineOptions.cs]: Build with the specified ID

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/GetLatestBuildOperation.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        GetLatestBuildCommandLineOptions _options;
+        public GetLatestBuildOperation(GetLatestBuildCommandLineOptions options)
+            : base(options)
+        {
+            _options = options;
+        }
+
+        /// <summary>
+        /// Adds a new channel with the specified name.
```
Comment (riarenas) [inline src/Microsoft.DotNet.Darc/src/Darc/Operations/GetLatestBuildOperation.cs]: "Gets the latest build of a repository", or something like that

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/GetLatestBuildOperation.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+                bool foundBuilds = false;
+                foreach (string possibleRepo in possibleRepos)
+                {
+                    foreach (Channel channel in channels)
+                    {
+                        Build latestBuild = await remote.GetLatestBuildAsync(possibleRepo, channel.Id);
+                        if (latestBuild != null)
+                        {
+                            if (foundBuilds)
```
Comment (riarenas) [inline src/Microsoft.DotNet.Darc/src/Darc/Operations/GetLatestBuildOperation.cs]: !foundBuilds

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/GetBuildOperation.cs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+                {
+                    Console.WriteLine("Could not find build with id '{_options.Id}'");
+                    return Constants.ErrorCode;
+                }
+
+                return Constants.SuccessCode;
+            }
+            catch (Exception e)
+            {
+                Logger.LogError(e, "Error: Failed to retrieve latest build.");
```
Comment (jcagme) [inline src/Microsoft.DotNet.Darc/src/Darc/Operations/GetBuildOperation.cs]:
```suggestion
                Logger.LogError(e, $"Error: Failed to retrieve build with id {_options.Id}.");
```

File: src/Microsoft.DotNet.Darc/src/Darc/Options/GetLatestBuildCommandLineOptions.cs [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using CommandLine;
+using Microsoft.DotNet.Darc.Operations;
+
+namespace Microsoft.DotNet.Darc.Options
+{
+    [Verb("get-latest-build", HelpText = "Retrieves the latest builds matching the specified criteria. " +
+        "If more than one build matches then multiple build are returned.")]
```
Comment (riarenas) [inline src/Microsoft.DotNet.Darc/src/Darc/Options/GetLatestBuildCommandLineOptions.cs]: [nit] multiple builds

**Pull request merged**